Source/processcontainers/implementations/CRunImplementation/CRunImplementation.cpp

```diff
@@ -0,0 +1,460 @@
+            while (_containers.size() > 0) {
+                _containers.back()->Release();
+                _containers.pop_back();
+            }
+        }
+    }
+
+    void CRunContainerAdministrator::Logging(const string& logPath, const string& loggingOptions)
+    {
+        // Only container-scope logging
+    }
+
+    CRunContainerAdministrator::ContainerIterator CRunContainerAdministrator::Containers()
+    {
+        return ContainerIterator(_containers);
+    }
+
+    void CRunContainerAdministrator::AddRef() const
+    {
+        _refCount++;
+    }
+
+    uint32_t CRunContainerAdministrator::Release()
+    {
+        --_refCount;
+
+        return (Core::ERROR_NONE);
+    }
+    
+    // Container
+    // ------------------------------------
+    CRunContainer::CRunContainer(string name, string path, string logPath)
+        : _refCount(1)
+        , _created(false)
+        , _name(name)        
+        , _bundle(path)
+        , _configFile(path + "/config.json")
+        , _logPath(logPath)
+        , _container(nullptr)
+        , _context()
+        , _pid()
+    {
+        // create a context
+        _context.bundle = _bundle.c_str();
+        _context.console_socket = nullptr;
+        _context.detach = true;
+        _context.fifo_exec_wait_fd = -1;
+        _context.force_no_cgroup = false;
+        _context.id = _name.c_str();
+        _context.no_new_keyring = true;
+        _context.no_pivot = false;
+        _context.no_subreaper = true;
+        _context.notify_socket = nullptr;
+        _context.output_handler = nullptr;
+        _context.output_handler_arg = nullptr;
+        _context.pid_file = nullptr;
+        _context.preserve_fds = 0;
+        _context.state_root = "/run/crun";
+        _context.systemd_cgroup = 0;
+
+        return;
+        if (logPath.empty() == false) { 
+            Core::Directory(logPath.c_str()).CreatePath();
+
+            libcrun_error_t error = nullptr;
+            int ret = init_logging (&_context.output_handler, &_context.output_handler_arg, _name.c_str(), (logPath + "/container.log").c_str(), &error);
+            if (ret != 0) {
+                TRACE_L1("Cannot initialize logging of container \"%s\" in directory %s\n. Error %d: %s", _name.c_str(), logPath.c_str(), error->status, error->msg);
+            }
+        }
+    }
+
+    CRunContainer::~CRunContainer() 
+    {
+
+    }  
+
+    const string CRunContainer::Id() const 
+    {
+        return _name;
+    }
+
+    uint32_t CRunContainer::Pid() const 
+    {
+        uint32_t result = 0;
+        libcrun_error_t error = nullptr;
+        libcrun_container_status_t status;
+        status.pid = 0;
+
+        result = libcrun_read_container_status (&status, _context.state_root, _name.c_str(), &error);
+        if (result != 0) {
+            TRACE_L1("Failed to get PID of container %s", _name.c_str());
+        }
+
+        return status.pid;
+    }
+
+    CRunContainer::MemoryInfo CRunContainer::Memory() const
+    {
```

Comment:
Memory measurements/aabstractions have already been part of the Core, why not use the core? This way enhancement (like sander did on the ProcessInfo object, will roll in for free)